vmm/src/api/http/http_endpoint.rs

```diff
@@ -220,6 +219,48 @@ impl PutHandler for VmAddNet {
 
 impl GetHandler for VmAddNet {}
 
+impl PutHandler for VmRestore {
+    fn handle_request(
+        &'static self,
+        api_notifier: EventFd,
+        api_sender: Sender<ApiRequest>,
+        body: &Option<Body>,
+        mut files: Vec<File>,
+    ) -> std::result::Result<Option<Body>, HttpError> {
+        if let Some(body) = body {
+            let mut restore_cfg: RestoreConfig = serde_json::from_slice(body.raw())?;
+
+            let mut fds = Vec::new();
+            if !files.is_empty() {
+                fds = files.drain(..).map(|f| f.into_raw_fd()).collect();
+            }
+            let expected_fds = match restore_cfg.net_fds {
+                Some(ref net_fds) => net_fds.iter().map(|net| net.num_fds).sum(),
+                None => 0,
+            };
+            if fds.len() != expected_fds {
+                return Err(HttpError::BadRequest);
```

Comment:
We need better error reporting here - see `error_response()` - also logging a message would be helpful.

Reply:
This returned `Err` is internally converted to a `error_response()` by the `handle_request` fn of `EndpointHandler` trait at https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/vmm/src/api/http/mod.rs#L108 .

The `handle_request` fn I added here is part of `impl PutHandler for Restore`. The return type of it should be `std::result::Result<Option<Body>, HttpError>`

Added a error!() log message.